stories/5-components/Helpers/AuFloatingUi.stories.js

```diff
@@ -0,0 +1,65 @@
+import { hbs } from 'ember-cli-htmlbars';
+
+export default {
+  title: 'Components/Helpers/AuFloatingUI',
+  argTypes: {
+    defaultPlacement: {
+      control: 'select',
+      options: [
+        'bottom-start',
+        'bottom',
+        'bottom-end',
+        'top-start',
+        'top',
+        'top-end',
+      ],
+      description: 'Set the default placement of the floater',
+    },
+    options: {
+      control: 'object',
+      description: 'Set general options of the floater & arrow',
+      defaultValue: {},
+    },
+    middleware: {
+      control: 'object',
+      description: 'Set specific floating-ui options',
+      defaultValue: [],
+    },
+  },
+  parameters: {
+    layout: 'padded',
+  },
+};
+
+const Template = (args) => ({
+  template: hbs`
+  <div class="sb-decorator-example-scroller" data-example-scroller>
+    <div class="sb-decorator-example-scroller-spacer">
```

Comment:
@Dietr I presume this can be converted to a Storybook add-on? Can I give it a go or do you already have some experience in this area?

In addition to this set-up, I would also like to add some functionality to automatically center the reference element within the container. Something like this:

```js
 const scroller = document.querySelector('[data-example-scroller]');

  if (scroller) {
    const child = scroller.children[0];

    const scrollerDimensions = {
      width: scroller.offsetWidth,
      height: scroller.offsetHeight,
    };

    const childDimensions = {
      width: child.offsetWidth,
      height: child.offsetHeight,
    };

    if (childDimensions.width > scrollerDimensions.width) {
      scroller.scrollLeft = (childDimensions.width - scrollerDimensions.width) / 2;
    }

    if (childDimensions.height > scrollerDimensions.height) {
      scroller.scrollTop = (childDimensions.height - scrollerDimensions.height) / 2;
    }
  }
```

Reply:
I don't have experience with writing storybook addons. I kind of bypass this by writing demo components in the ember test application.

Reply:
Thanks for the advice! I've put the extracted functionality inside: 
- tests/dummy/app/components/au-example-scroller.hbs
- tests/dummy/app/components/au-example-scroller.js